dns/name.py

```diff
@@ -843,6 +848,40 @@ def parent(self) -> "Name":
             raise NoParent
         return Name(self.labels[1:])
 
+    def predecessor(self, origin: "Name", prefix_ok: bool = True) -> "Name":
+        """Return the maximal predecessor of *name* in the DNSSEC ordering in the zone
+        whose origin is *origin*, or return the longest name under *origin* if the
+        name is origin (i.e. wrap around to the longest name, which may still be
+        *origin* due to length considerations.
+
+        The relativity of the name is preserved, so if this name is relative
+        then the method will return a relative name, and likewise if this name
+        is absolute then the predecessor will be absolute.
+
+        *prefix_ok* indicates if prefixing labels is allowed, and
+        defaults to ``True``.  Normally it is good to allow this, but if computing
+        a maximal predecessor at a zone cut point then ``False`` must be specified.
+        """
+        return _predecessor(self, origin, prefix_ok)
```

Comment:
Is there a reason for this extra level of indirection?

Reply:
Yes, this was excessive.  I got rid of that level of indirection, though I haven't tried to smoosh everything into the Name class.  I feel like most of the predecessor/successor code is better as helper functions as it only works on absolute names.